src/Tokenizer/Tokens.php

```diff
@@ -696,6 +744,46 @@ public function getPrevMeaningfulToken(int $index): ?int
         return $this->getMeaningfulTokenSibling($index, -1);
     }
 
+    /**
+     * Get index for closest next token that is not a whitespace or comment.
+     *
+     * @param int $index token index
+     *
+     * @throws \LogicException when no such index exists
+     *
+     * @internal
+     */
+    public function meaningfulTokenIndexAfter(int $index): int
```

Comment:
case 5

I wonder if we should make new, alternative method here, or adjust behaviour of original method.
imagine we deprecate and later remove old method - all fixers will need to start using other method. including the custom fixers.
maybe we can groom behaviour of existing method behind toggle [aka `legacyMode`] and keep same method in place, trigger deprecation warning when working in `legacyMode` and exception otherwise?

Reply:
ref https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/blob/v2.19.3/src/Tokenizer/Tokens.php#L96

Reply:
I prefer multiple specific functions over a single one with flag parameters because it's more readable and often more type-safe. But the main reason for me to add a new method was that the name of the current method has always been a bit confusing to me: when reading `get…Token()`, I think it will return `Token|null` while it actually returns `int|null`. I wanted to prevent that with the new name.

Reply:
Imho:

we have `method_A` now, that works in given way.

We should align how we would like to have it working, with a way that we can easily migrate codebase [eg migration from null-return to exception-raised is relatively easy, as we don't expect missing token in majority of places ; yet migration from multi-kind needle to single-kind needle will be a massing change to apply]. Then, we should make the method trigger warnings [not exception] if someone would use it in a way that will work differently in next version of method. And then, we could have compact toggle that is raising the warning or actually changing the logic [eg raising exception instead of returning null]. I believe as long as we don't change the single/multi-kind needle, we do not need to change prototype of the method.